libcaf_core/caf/scheduler.test.cpp

```diff
@@ -0,0 +1,162 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/scheduler.hpp"
+
+#include "caf/test/outline.hpp"
+#include "caf/test/scenario.hpp"
+
+#include "caf/actor_system_config.hpp"
+#include "caf/async/file.hpp"
+#include "caf/detail/latch.hpp"
+
+#include <string>
+
+using namespace caf;
+using namespace std::literals;
+using detail::latch;
+
+namespace {
+
+struct testee : resumable, ref_counted {
+  testee(std::shared_ptr<latch> latch_handle)
+    : rendesvous(std::move(latch_handle)) {
+  }
+  ~testee() = default;
+  subtype_t subtype() const override {
+    return resumable::function_object;
+  }
+  resume_result resume(execution_unit*, size_t max_throughput) override {
+    if (++runs == 10) {
+      rendesvous->count_down();
+      received_throughput = max_throughput;
+      return resumable::done;
+    }
+    return resumable::resume_later;
+  }
+  void intrusive_ptr_add_ref_impl() override {
+    ref();
+  }
+  void intrusive_ptr_release_impl() override {
+    deref();
+  }
+  std::atomic<size_t> runs = 0;
+  size_t received_throughput = 0;
+  std::shared_ptr<latch> rendesvous;
+};
+
+OUTLINE("scheduling resumables") {
+  GIVEN("an actor system using the work <sched> scheduler") {
+    auto sched = block_parameters<std::string>();
+    actor_system_config cfg;
+    cfg.set("caf.scheduler.policy", sched);
+    cfg.set("caf.scheduler.max-throughput", 5);
+    actor_system sys(cfg);
+    auto& sut = sys.scheduler();
+    WHEN("scheduling a resumable") {
+      auto rendesvous = std::make_shared<latch>(2);
+      auto worker = make_counted<testee>(rendesvous);
+      worker->ref();
+      sut.enqueue(worker.get());
+      THEN("expect the resumable to be executed until done") {
+        rendesvous->count_down_and_wait();
+        check_eq(worker->runs.load(), 10u);
+      }
+      AND_THEN("expect the correct max throughput") {
+        check_eq(worker->received_throughput, 5u);
+      }
+      AND_THEN("the scheduler releases the ref when done") {
+        check_eq(worker->get_reference_count(), 1u);
+      }
+    }
+    AND_WHEN("scheduling multiple resumables") {
```

Comment:
Has nothing to do with the previous test, so it shouldn't continue it.

```suggestion
    WHEN("scheduling multiple resumables") {
```

Reply:
Opened #1776 for this. I get an exception when I put two WHEN blocks inside an outline.